docs/design-docs/cli/cli-ux.md

```diff
@@ -0,0 +1,378 @@
+# Epiphany CLI UX
+
+Affected version: unknown
+
+## Goals
+
+This document aim is to improve user experience with epicli tool with strong emphasis to lower entry level for new users. It provides idea for following scenarios: 
+ * epicli installation
+ * environment initialization and deployment
+ * environment component update
+ * cli tool update
+ * add component to existing environment
+ 
+## Assumptions
+
+Following scenarios assume: 
+ * there is component version introduced - epicli version is separated from component version. It means that i.e. epicli v0.0.1 can provide component PostgreSQL 10.x and/or PostgreSQL 11.x.
+ * there is server-side component - epiphany environment is always equipped with server side daemon component exposing some API to epicli. 
+ 
+## Convention
+I used square brackets with dots inside: 
+```
+[...]
+```
+to indicate processing or some not important for this document output. 
+
+## Story
+
+### epicli installation
+
+To increase user base we need to provide brew formulae to allow simple installation. 
+```
+> brew install epicli
+``` 
+ 
+### environment initialization and deployment
+
+#### init
+
+As before user should be able to start interaction with epicli with `epicli init` command. In case of no parameters interactive version would be opened. 
```

Comment:
Jhipster has really nice interactive mode. We can try to achieve something similar to their implementation.